threshold_time : 0.5
twist_threshold: 0.2
